### PR TITLE
Remove references to "cam" in submodule paths comment strings

### PR DESCRIPTION
**Pull request by singhbalwinder**
Replace references to "cam" and shorten submodule names in .gitmodules file

### CODE REVIEW

**Review by jgfouca — APPROVED** [review]

File: .gitmodules [hunk 1]
```diff
@@ -15,7 +15,7 @@
 	path = externals/kokkos
 	url = git@github.com:E3SM-Project/kokkos.git
 	branch = e3sm/kokkos
-[submodule "components/cam/src/physics/rrtmgp/external"]
+[submodule "RRMGP"]
```
Comment (sarats) [inline .gitmodules:18]: Just eyeballing, isn't the 'T' missing in RRTMGP here?

Comment (sarats) [reply .gitmodules:18]: Or does this not matter in terms of finding submodule....

Comment (rljacob) [reply .gitmodules:18]: Not for finding since what is in quotes is just a label to use for submodule commands.  But the T should be there.

Comment (singhbalwinder) [reply .gitmodules:18]: Thanks for noticing the typo. I will fix it.

**Pull request merged**